src/FluentCheck.ts

```diff
@@ -175,58 +175,26 @@ class FluentCheckUniversal<K extends string, A, Rec extends ParentRec & Record<K
     for (const tp of collection) {
       testCase[this.name] = tp
       const result = callback(testCase)
-      if (!result.satisfiable) {
+      if (result.satisfiable === this.breakCondition) {
         result.addExample(this.name, tp)
         return this.run(testCase, callback, result, depth + 1)
       }
     }
 
     return example
   }
-}
-
-class FluentCheckExistential<K extends string, A, Rec extends ParentRec & Record<K, A>, ParentRec extends {}>
-  extends FluentCheck<Rec, ParentRec> {
-
-  private cache: Array<FluentPick<A>>
-  private dedup: Arbitrary<A>
-
-  constructor(
-    protected readonly parent: FluentCheck<ParentRec, any>,
-    public readonly name: K,
-    public readonly a: Arbitrary<A>,
-    config: FluentConfig) {
-
-    super(parent, config)
-    this.dedup = a.unique()
-    this.cache = this.dedup.sampleWithBias(this.configuration.sampleSize)
-  }
-
-  protected run(
-    testCase: FluentPicks,
-    callback: (arg: FluentPicks) => FluentResult,
-    partial: FluentResult | undefined = undefined,
-    depth = 0): FluentResult {
 
-    const example = partial || new FluentResult(false)
-
-    const collection = depth === 0 ?
-      this.cache :
-      (partial !== undefined ?
-        this.dedup.shrink(partial.example[this.name]).sampleWithBias(this.configuration.shrinkSize) :
-        [])
+  abstract breakCondition: boolean
+}
 
-    for (const tp of collection) {
-      testCase[this.name] = tp
-      const result = callback(testCase)
-      if (result.satisfiable) {
-        result.addExample(this.name, tp)
-        return this.run(testCase, callback, result, depth + 1)
-      }
-    }
+class FluentCheckUniversal<K extends string, A, Rec extends ParentRec & Record<K, A>, ParentRec extends {}>
+  extends FluentCheckQuantifier<K, A, Rec, ParentRec> {
+  breakCondition = false
```

Comment:
Now that it's a boolean I prefer if it is called `breakValue` as suggested rather than `breakCondition`.